src/main/drivers/accgyro/accgyro_spi_icm40609.c

```diff
@@ -0,0 +1,769 @@
+    ICM40609_UI_FILT_ORDER_3RD = 2
+} icm40609UiFiltOrder_e;
+
+typedef enum {
+    ICM40609_HPF_ORDER_1ST = 0,
+    ICM40609_HPF_ORDER_2ND = 1
+} icm40609HpfOrder_e;
+
+// Bandwidth selection for High-Pass Filter
+// ICM40609_REG_GYRO_CONFIG_STATIC10 - 0x13 bits [3:1]
+// NOTE: clarify with new datasheet, section 5.6 not found in V1.2
+typedef enum {
+    ICM40609_HPF_BW_0 = 0,
+    ICM40609_HPF_BW_1 = 1,
+    ICM40609_HPF_BW_2 = 2,
+    ICM40609_HPF_BW_3 = 3,
+    ICM40609_HPF_BW_4 = 4,
+    ICM40609_HPF_BW_5 = 5,
+    ICM40609_HPF_BW_6 = 6,
+    ICM40609_HPF_BW_7 = 7,
+} icm40609HpfBw_e;
+
+// Bandwidth selection for Notch Filter GYRO_NF_BW_SEL
+// ICM40609_REG_GYRO_CONFIG_STATIC10 - 0x13 bits [6:4]
+typedef enum {
+    ICM40609_GYRO_NF_BW_1449HZ = 0,
+    ICM40609_GYRO_NF_BW_680HZ  = 1,
+    ICM40609_GYRO_NF_BW_329HZ  = 2,
+    ICM40609_GYRO_NF_BW_162HZ  = 3,
+    ICM40609_GYRO_NF_BW_80HZ   = 4,
+    ICM40609_GYRO_NF_BW_40HZ   = 5,
+    ICM40609_GYRO_NF_BW_20HZ   = 6,
+    ICM40609_GYRO_NF_BW_10HZ   = 7,
+} icm40609GyroNfBw_e;
+
+typedef enum {
+    ICM40609_TEMP_FILT_BW_4000HZ = 0, // 4000Hz, 0.125ms latency (default)
+    ICM40609_TEMP_FILT_BW_170HZ  = 1, // 170Hz, 1ms latency
+    ICM40609_TEMP_FILT_BW_82HZ   = 2, // 82Hz, 2ms latency
+    ICM40609_TEMP_FILT_BW_40HZ   = 3, // 40Hz, 4ms latency
+    ICM40609_TEMP_FILT_BW_20HZ   = 4, // 20Hz, 8ms latency
+    ICM40609_TEMP_FILT_BW_10HZ   = 5, // 10Hz, 16ms latency
+    ICM40609_TEMP_FILT_BW_5HZ    = 6, // 5Hz, 32ms latency
+} icm40609TempFiltBw_e;
+
+static void icm40609SelectUserBank(const extDevice_t *dev, uint8_t bank)
+{
+    if (bank > 4) return; // out of range
+    spiWriteReg(dev, ICM40609_REG_BANK_SEL, bank & 0x07); // bit [2:0]
+}
+
+static void setGyroAccPowerMode(const extDevice_t *dev, bool enable)
+{
+    icm40609SelectUserBank(dev, ICM40609_USER_BANK_0);
+
+    spiWriteReg(dev, ICM40609_REG_PWR_MGMT0,
+        ICM40609_RC_ON |
+        ICM40609_TEMP_SENSOR_ENABLED |
+        (enable ? ICM40609_GYRO_MODE_LN | ICM40609_ACCEL_MODE_LN
+                : ICM40609_GYRO_MODE_OFF | ICM40609_ACCEL_MODE_OFF));
+
+    if (enable) {
+        delayMicroseconds(200);
+    } else {
+        delay(50);
+    }
+}
+
+static void icm40609GetAafParams(uint16_t targetHz, uint8_t *delt, uint8_t *deltsqr, uint8_t *bitshift)
+{
+    *delt = aafProfiles[ICM40609_AAF_PROFILE_COUNT - 1].delt;
+    *deltsqr = aafProfiles[ICM40609_AAF_PROFILE_COUNT - 1].deltsqr;
+    *bitshift = aafProfiles[ICM40609_AAF_PROFILE_COUNT - 1].bitshift;
+
+    for (int i = 0; i < ICM40609_AAF_PROFILE_COUNT; i++) {
+        if (targetHz <= aafProfiles[i].hz) {
+            *delt = aafProfiles[i].delt;
+            *deltsqr = aafProfiles[i].deltsqr;
+            *bitshift = aafProfiles[i].bitshift;
+            break;
+        }
+    }
+}
+
+static void icm40609SetAccelAafByHz(const extDevice_t *dev, bool aafEnable, uint16_t targetHz)
+{
+    icm40609SelectUserBank(dev, ICM40609_USER_BANK_2);
```

Comment:
This is very error prone. It may be better to use `(bank << 8) | reg` a register number and (optionally) cache last selected bank.
Some overhead is fine for initialization, after it direct read should do.